nixos/modules/programs/wayland/hyprland.nix

```diff
@@ -38,12 +38,12 @@ in
     xwayland.enable = lib.mkEnableOption "XWayland" // { default = true; };
 
     systemd.setPath.enable = lib.mkEnableOption null // {
-      default = true;
+      default = lib.versionOlder cfg.package.version "0.41.2";
```

Comment:
```suggestion
      defaultText = literalExpression ''lib.versionOlder cfg.package.version "0.41.2"'';
```

I think you need to also add defaultText here, because default is a complex expression.

Reply:
I managed to build the manual by using 

```
      default = lib.versionOlder cfg.package.version "0.41.2";
      defaultText = "Hello World";
```

![image](https://github.com/NixOS/nixpkgs/assets/50095635/240b2551-b2cd-40ad-8645-3f506f6ebe09)

I assume default is not ignored and defaultText is only for the docs?

Reply:
Oops sorry, my suggestion is not correct.

```suggestion
      default = lib.versionOlder cfg.package.version "0.41.2";
      defaultText = literalExpression ''lib.versionOlder cfg.package.version "0.41.2"'';
```

default and defaultText should be both here.
defaultText is only textual representation for the manual.

See https://nixos.org/manual/nixos/stable/#sec-option-declarations

> A textual representation of the default value to be rendered verbatim in the manual. Useful if the default value is a complex expression or depends on other values or packages. Use lib.literalExpression for a Nix expression, lib.literalMD for a plain English description in [Nixpkgs-flavored Markdown](https://nixos.org/nixpkgs/manual/#sec-contributing-markup) format.

Reply:
Instead of something like `defaultText = literalExpression ''lib.versionOlder cfg.package.version "0.41.2"'';`

maybe it would be better to use a english description in markdown format?

Reply:
Done, thanks!